greenplumpython/group.py

```diff
@@ -0,0 +1,16 @@
+from typing import TYPE_CHECKING, Callable, Iterable
+
+if TYPE_CHECKING:
+    from .expr import Expr
+    from .func import FunctionExpr
+    from .table import Table
+
+
+class TableRowGroup:
+    def __init__(self, table: "Table", group_by: Iterable["Expr"]) -> None:
+        self._table = table
+        self._group_by = [self._table[e] if isinstance(e, str) else e for e in group_by]
+
+    def aggregate(self, func: Callable, *args: "Expr") -> "FunctionExpr":
+        qualified_args = [self._table[e] if isinstance(e, str) else e for e in args]
+        return func(*qualified_args, group_by=self._group_by)
```

Comment:
As a syntax sugar, this doesn't look too bad IMO.

My concern is , sooner or later, we need to define the built-in aggregates/functions like:

```
# gp/aggregates/defs.py
def count():
    return gp.aggregate("count")
```

so probably we need a proxy method here like:

```
class TableRowGroup:
    def count():
        return aggregate(gp.aggregates.count);
```

Since to create the `count` aggregate, a `db` obj needs to be passed as an argument. Then how do we define those built-in functions/aggregates?

Reply:
> count() is already available in every db and we don't need to create it.

I know

> In fact, every "built-in" SQL functions that we want to create a Python method for should already be available in every db. That's what "built-in" means.

Exactly, how do we ship a `count()` method by default? How does the `def count()` look like in our code base?